core/lib/types/CRing.agda

```diff
@@ -0,0 +1,89 @@
+{-# OPTIONS --without-K --rewriting #-}
+
+open import lib.Basics
+open import lib.types.Group
+open import lib.groups.Homomorphism
+
+module lib.types.CRing where
+
+-- 1-approximation of commutative rings without higher coherence conditions.
+record CRingStructure {i} (El : Type i)
+  : Type i where
+  constructor ring-structure
+  field
+    add-group-struct : GroupStructure El
+
+  open GroupStructure add-group-struct
+
+  zero : El
+  zero = ident
+
+  add : El → El → El
+  add = comp
+
+  field
+    add-comm    : ∀ a b → add a b == add b a
+    one         : El
+    mult        : El → El → El
+    mult-unit-l : ∀ a → mult one a == a
+    mult-assoc  : ∀ a b c → mult (mult a b) c == mult a (mult b c)
+    mult-comm   : ∀ a b → mult a b == mult b a
+    distr-l : ∀ a b c → mult (add a b) c == add (mult a c) (mult b c)
```

Comment:
I wonder if having commutative monoids would help... but we can think about the refactoring later.